**Aplicaciones en ciencia de datos**

**Laboratorio 3**

María Paula Angulo Collantes - 202212044

Danna Gabriela Castro Villalba - 202615188

In [2]:
import pandas as pd

**Punto 1**

In [4]:
#Carga
df = pd.read_csv('estudiantes - Copy.csv')

In [5]:
#Primeras filas
df.head()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
0,E0001,Economía,5,21,Mujer,16.8,5.5,61.0,Bus,4,4,3,7.2,A veces,3,Ingeniería de datos,Presencial,3.47,2026-08-01
1,E0002,Administración,6,23,Mujer,23.5,7.2,98.0,TransMilenio,4,4,2,10.8,Nunca,2,No estoy seguro,Presencial,3.69,2026-08-10
2,E0003,Administración,1,17,Hombre,12.6,6.7,55.0,TransMilenio,5,4,3,6.8,A veces,4,Analítica de negocio,Presencial,3.26,2026-08-21
3,E0004,Ciencia de Datos,8,21,Mujer,15.7,6.1,57.0,Moto,4,3,1,14.3,Nunca,4,Analítica de negocio,Presencial,2.90,2026-08-19
4,E0005,Diseño,3,22,Mujer,20.5,6.4,50.0,Carro,5,4,5,11.0,A veces,2,No estoy seguro,Presencial,3.54,2026-08-04


In [8]:
#Últimas filas
df.tail()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
2507,E2113,Ciencia de Datos,1,18,Mujer,13.7,6.3,81.0,Caminando,5,4,5,9.7,Frecuentemente,3,Analítica de negocio,Presencial,3.67,2026-08-10
2508,E0456,Administración,2,19,Mujer,17.7,5.8,47.0,Caminando,4,4,3,4.9,Frecuentemente,1,IA y ML,Presencial,3.24,2026-08-04
2509,E1001,Ingeniería de Sistemas,1,19,Mujer,20.1,5.2,92.0,Caminando,5,3,1,9.1,Frecuentemente,4,Analítica de negocio,Presencial,3.82,2026-08-19
2510,E0438,Ingeniería Industrial,1,19,Mujer,15.5,7.8,50.0,TransMilenio,3,4,5,3.7,Todos los días,3,Analítica de negocio,Presencial,3.79,2026-08-11
2511,E1192,Administración,2,20,Mujer,15.6,7.7,50.0,Caminando,3,4,6,10.8,A veces,4,Visualización,Presencial,3.75,2026-08-02


In [10]:
#Dimensiones del DataFrame
df.shape

(2512, 19)

In [11]:
#Nombre de las columnas
df.columns

Index(['student_id', 'programa', 'semestre', 'edad', 'genero',
       'horas_estudio_semana', 'horas_sueno_noche',
       'tiempo_desplazamiento_min', 'medio_transporte', 'nivel_estres',
       'satisfaccion_academica', 'dias_ejercicio_semana',
       'horas_sociales_semana', 'uso_ia_generativa', 'confianza_programando',
       'area_interes_datos', 'modalidad_preferida', 'promedio_acumulado',
       'fecha_encuesta'],
      dtype='str')

In [12]:
#Tipos de datos inferidos por Pandas
df.dtypes

student_id                       str
programa                         str
semestre                       int64
edad                           int64
genero                           str
horas_estudio_semana         float64
horas_sueno_noche            float64
tiempo_desplazamiento_min    float64
medio_transporte                 str
nivel_estres                   int64
satisfaccion_academica         int64
dias_ejercicio_semana          int64
horas_sociales_semana        float64
uso_ia_generativa                str
confianza_programando          int64
area_interes_datos               str
modalidad_preferida              str
promedio_acumulado           float64
fecha_encuesta                   str
dtype: object

**Preguntas**

a. ¿Qué representa una fila del dataset?

**Rta:** Cada fila del dataset representa a un estudiante individual. Cada registro está identificado por un código único y describe detalladamente las respuestas de un único alumno a la encuesta.

b. ¿Cuál es la unidad de análisis?

**Rta:** La unidad de análisis es el estudiante.

c. ¿Cuántas observaciones y variables hay?

**Rta:** Hay 2512 observaciones (filas), cada una de ellas corresponde a un estudiante único encuestado y 19 variables (columnas), cada una representa una característica medida o registrada de los estudiantes.

d. ¿Qué parece representar el dataset en términos del problema o dominio?

**Rta:** En términos del problema y su dominio, este dataset representa un diagnóstico integral y multidimensional de la vida y el bienestar de la comunidad estudiantil en un entorno de educación superior.

**Punto 2**

In [23]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2512 entries, 0 to 2511
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   student_id                 2512 non-null   str    
 1   programa                   2512 non-null   str    
 2   semestre                   2512 non-null   int64  
 3   edad                       2512 non-null   int64  
 4   genero                     2512 non-null   str    
 5   horas_estudio_semana       2512 non-null   float64
 6   horas_sueno_noche          2440 non-null   float64
 7   tiempo_desplazamiento_min  2456 non-null   float64
 8   medio_transporte           2512 non-null   str    
 9   nivel_estres               2512 non-null   int64  
 10  satisfaccion_academica     2512 non-null   int64  
 11  dias_ejercicio_semana      2512 non-null   int64  
 12  horas_sociales_semana      2512 non-null   float64
 13  uso_ia_generativa          2512 non-null   str    
 14  con

In [25]:
df.nunique()

student_id                   2500
programa                        8
semestre                       10
edad                           13
genero                          5
horas_estudio_semana          305
horas_sueno_noche              57
tiempo_desplazamiento_min     135
medio_transporte                7
nivel_estres                    5
satisfaccion_academica          4
dias_ejercicio_semana           8
horas_sociales_semana         189
uso_ia_generativa               4
confianza_programando           5
area_interes_datos              6
modalidad_preferida             3
promedio_acumulado            191
fecha_encuesta                 21
dtype: int64

| Variables                 | Tipo en Python | Escala   | Descripción                                             | Ejemplo             | Observaciones                                                                                                     |
|:-------------------------:|:--------------:|:--------:|:-------------------------------------------------------:|:-------------------:|:-----------------------------------------------------------------------------------------------------------------:|
| student_id                | str            | Nominal  | Identificador único del estudiante                      | E0001               | No es único                                                                                                       |
| programa                  | str            | Nominal  | Programa académico del estudiante                       | Economía            |                                                                                                                   |
| semestre                  | int64          | Ordinal  | Semestre que cursa el estudiante                        | 5                   | Va de 1 - 10                                                                                                      |
| edad                      | int64          | Discreta | Edad del estudiante en años                             | 21                  | Va de 16 - 28                                                                                                     |
| genero                    | str            | Nominal  | Genero con el que se identifica el estudiante           | Mujer               |                                                                                                                   |
| horas_estudio_semana      | float64        | Continua | Horas que dedica a estudiar                             | 16.8                | Rango observado: 0 a 95 horas; el valor máximo (95) es alto — verificar si es un dato real o un error de captura. |
| horas_sueno_noche         | float64        | Continua | Horas que duerme por la noche                           | 5.5                 | 72 valores nulos                                                                                                  |
| tiempo_desplazamiento_min | float64        | Continua | Tiempo de desplazamiento al lugar de estudio en minutos | 61.0                | 56 valores nulos. Valor máximo de 360 min (6 horas) posible outlier o error de captura                            |
| medio_transporte          | str            | Nominal  | Medio de transporte usado                               | Bus                 |                                                                                                                   |
| nivel_estres              | int64          | Ordinal  | Nivel de estrés autopercibido                           | 4                   | Va de 1 - 5                                                                                                       |
| satisfaccion_academica    | int64          | Ordinal  | Nivel de satisfacción con la vida académica             | 4                   | Va de 1 - 5                                                                                                       |
| dias_ejercicio_semana     | int64          | Discreta | Número de días a la semana que hace ejercicio           | 3                   | Va de 0 - 7                                                                                                       |
| horas_sociales_semana     | float64        | Continua | Número de horas a la semana que dedica a socializar     | 7.2                 | Va de 0 - 22.7                                                                                                    |
| uso_ia_generativa         | str            | Ordinal  | Frecuencia de herramientas de IA                        | A veces             | Categorías: Nunca < A veces < Frecuentemente < Todos los días                                                     |
| confianza_programando     | int64          | Ordinal  | Nivel de confianza autopercibido en programación        | 3                   | Va de 1 - 5                                                                                                       |
| area_interes_datos        | str            | Nominal  | Área de interés dentro de ciencia de datos              | Ingeniería de datos | 41 Valores nulos                                                                                                  |
| modalidad_preferida       | str            | Nominal  | Modalidad de estudio preferida                          | Presencial          | Categorías: Presencial, Híbrida, Virtual.                                                                         |
| promedio_acumulado        | float64        | Continua | Promedio académico acumulado del estudiante             | 3.47                | Va de 2.36 - 4.73                                                                                                 |
| fecha_encuesta            | str            | -        | Fecha en la que se respondio la encuenta                | 2026-08-01          | Está almacenada como texto (str), no como tipo datetime                                                           |


**Punto 3**

In [41]:
#Dimensiones 
print("Dimensiones del DataFrame:", df.shape)
#Tipos
print("Tipos de datos de las columnas:\n",
      
      df.dtypes,

      "\nSin embargo,fecha_encuesta se infirió como str en lugar de un tipo de fecha")
#Faltantes
print("Los valores faltantes se encuentran en las columnas: horas_sueno_noche, tiempo_desplazamiento, area_interes_datos")
#Duplicados
print("Hay",df.duplicated().sum(),"filas duplicadas en el DataFrame")
#Cardinalidad
print("Cardinalidad de las columnas:\n",df.nunique())
#Distribución
print("Distribución de las columnas:\n", df.describe())

Dimensiones del DataFrame: (2512, 19)
Tipos de datos de las columnas:
 student_id                       str
programa                         str
semestre                       int64
edad                           int64
genero                           str
horas_estudio_semana         float64
horas_sueno_noche            float64
tiempo_desplazamiento_min    float64
medio_transporte                 str
nivel_estres                   int64
satisfaccion_academica         int64
dias_ejercicio_semana          int64
horas_sociales_semana        float64
uso_ia_generativa                str
confianza_programando          int64
area_interes_datos               str
modalidad_preferida              str
promedio_acumulado           float64
fecha_encuesta                   str
dtype: object 
Sin embargo,fecha_encuesta se infirió como str en lugar de un tipo de fecha
Los valores faltantes se encuentran en las columnas: horas_sueno_noche, tiempo_desplazamiento, area_interes_datos
Hay 12 filas duplicad

**Punto 4:** ¿Hay columnas cuyo significado o nombre no sea suficientemente claro?

Sí, se encontraron varias. Por ejemplo, en nivel_estres, satisfaccion_academica y confianza_programando el nombre no especifica qué significa cada extremo de la escala (no queda claro si 1 representa "bajo" y 5 "alto", o al revés). En tiempo_desplazamiento_min no se aclara si el desplazamiento es hacia la universidad o hacia otro lugar.

**Punto 5:** ¿Existen códigos, categorías o unidades que requerirían documentación adicional?

Sí. El campo student_id usa un código tipo "E0001" cuyo formato o lógica de asignación no está documentado (no se sabe si el número corresponde a un orden de inscripción, un consecutivo aleatorio, u otro criterio). Las escalas numéricas de nivel_estres, satisfaccion_academica y confianza_programando tampoco cuentan con documentación sobre qué representa cada valor del 1 al 5, lo cual dificulta interpretar correctamente los resultados. La unidad de tiempo_desplazamiento_min se puede inferir por el sufijo "_min" (minutos), pero no se especifica si corresponde a un trayecto de ida, de ida y vuelta, o si varía según el día de la semana. La categoría "No estoy seguro" en area_interes_datos también requeriría documentación adicional para precisar si debe tratarse como una respuesta válida o como un valor faltante. Por último, no hay documentación sobre la metodología o el rango completo de promedio_acumulado (por ejemplo, si la escala oficial va de 0.0 a 5.0, y qué unidad académica lo calcula y reporta).

**Punto 6:** ¿Hay variables cuyo tipo en Python no corresponda con su significado analítico?

Sí, se identificaron algunos casos. fecha_encuesta está almacenada como str en lugar de un tipo de fecha (datetime), lo cual impide realizar operaciones temporales directas (como calcular diferencias de días u ordenar cronológicamente) sin antes convertirla con pd.to_datetime(). De forma similar, semestre está almacenado como int64, pero conceptualmente es una variable ordinal que indica progresión académica más que una cantidad que tenga sentido sumar o promediar de forma directa.

**Punto 7** ¿Detectan categorias inconsistentes, rangos improbables o posibles errores de captura?

Sí, durante la revisión se identificaron algunas categorías inconsistentes y rangos que podrían considerarse improbables. Primero, se encontraron diferencias en la forma de registrar algunas categorías; elpor ejemplo, en la variable "género" aparecen tanto “Mujer” como “mujer”, que representan la misma categoría, pero son interpretadas como valores diferentes. Y pasa igual en una de las variables de programas donde hay "Ciencia de datos" y "Ciencia datos" representando el mismo programa pero como diferentes variables, y asi mismo con una variable del "medio de transporte",ya que hay registros con diferencias en la escritura de “Transmilenio”, por lo que sería necesario estandarizar estas categorías antes de realizar análisis. En cuanto a las variables numéricas, la mayoría presenta rangos coherentes. Sin embargo, hay algunos valores que podrían ir muy al extremo, como un máximo de 95 horas de estudio por semana, que son casi 13 horas diarias, y un tiempo máximo de desplazamiento de 360 minutos, o sea, 6 horas. Aunque estos valores parecen elevados, no se pueden considerar errores de una vez, ya que podrían ser a situaciones reales y deberían ser verificados. Por lo que es prudente estandarizar las categorías que son inconsistentes, revisar los valores extremos y confirmar aquellos datos que puedan ser considerados posibles errores.

In [11]:
df["programa"].value_counts()

programa
Ciencia de Datos          475
Ingeniería Industrial     453
Ingeniería de Sistemas    427
Economía                  383
Administración            362
Diseño                    209
Psicología                200
Ciencia Datos               3
Name: count, dtype: int64

In [7]:
df["genero"].value_counts(dropna=False)

genero
Hombre                   1238
Mujer                    1148
Prefiere no responder      72
No binario                 47
mujer                       7
Name: count, dtype: int64

In [8]:
df["medio_transporte"].value_counts(dropna=False)

medio_transporte
TransMilenio    788
Bus             501
Carro           442
Caminando       330
Bicicleta       252
Moto            188
Transmilenio     11
Name: count, dtype: int64

In [13]:
print(df.isnull().sum())

student_id                    0
programa                      0
semestre                      0
edad                          0
genero                        0
horas_estudio_semana          0
horas_sueno_noche            72
tiempo_desplazamiento_min    56
medio_transporte              0
nivel_estres                  0
satisfaccion_academica        0
dias_ejercicio_semana         0
horas_sociales_semana         0
uso_ia_generativa             0
confianza_programando         0
area_interes_datos           41
modalidad_preferida           0
promedio_acumulado            0
fecha_encuesta                0
dtype: int64


In [ ]:
# Ver los rangos min y mx de las categorias
columnas_numericas = [
    "semestre",
    "edad",
    "horas_estudio_semana",
    "horas_sueno_noche",
    "tiempo_desplazamiento_min",
    "nivel_estres",
    "satisfaccion_academica",
    "dias_ejercicio_semana",
    "horas_sociales_semana",
    "confianza_programando",
    "promedio_acumulado"
]

for columna in columnas_numericas:
    print(
        columna,
        "mínimo:", df[columna].min(),
        "máximo:", df[columna].max()
    )

semestre mínimo: 1 máximo: 10
edad mínimo: 16 máximo: 28
horas_estudio_semana mínimo: 0.0 máximo: 95.0
horas_sueno_noche mínimo: 3.6 máximo: 10.5
tiempo_desplazamiento_min mínimo: 0.0 máximo: 360.0
nivel_estres mínimo: 1 máximo: 5
satisfaccion_academica mínimo: 2 máximo: 5
dias_ejercicio_semana mínimo: 0 máximo: 7
horas_sociales_semana mínimo: 0.0 máximo: 22.7
confianza_programando mínimo: 1 máximo: 5
promedio_acumulado mínimo: 2.36 máximo: 4.73


**Punto 8**¿Que informacion adicional pedirıan al dueño de los datos antes de analizar?


En primer lugar, se preguntaría qué significa exactamente cada valor de las variables con escala, es decir, qué representa cada número de la escala del 1 al 5, como por ejemplo si en el "nivel_estres" 1 es un nivel bajo de estrés o alto. Por otro lado, sería importante confirmar cuál es la escala utilizada para "promedio_acumulado", por ejemplo, si corresponde a una escala de 0 a 5 y si todos los programas utilizan la misma. Para la variable "tiempo_desplazamiento_min", se pediría información sobre  si el tiempo registrado corresponde únicamente al trayecto de ida o al tiempo total de ida y regreso, pues esto permitiría interpretar correctamente valores extremos como los 360 minutos que se vieron. De igual manera, sería necesario conocer qué actividades fueron consideradas dentro de las "horas de estudio por semana" , es decir, si incluyen clases, tareas, lecturas, trabajos, proyectos o únicamente el estudio personal. Además, se consultoria por las filas duplicadas que se detallaron, con el fin de determinar si corresponden a encuestas repetidas o a errores de registro y evitar eliminarlas sin causa justa. También se solicitaría información sobre los valores faltantes, para saber si los campos vacíos significan que la persona no respondió, o si la pregunta no aplicaba o si el dato se perdió en esa recolección.  ¿Qué información adicional pediríamos al dueño de los datos


**Punto 9**¿Que riesgos tendrıa comenzar un analisis sin resolver estas dudas?


Comenzar el análisis sin resolver esas  dudas podría afectar la confiabilidad de los resultados y  la veracidad de ellos, pues puede llevarnos a tomar decisiones equivocadas sobre los datos. Las inconsistencias, los registros duplicados, los valores faltantes y los posibles valores extremos podrían alterar los cálculos, las gráficas, las estadísticas y generar conclusiones que no representen correctamente la población que se estudió. Además, no tener claridad sobre el significado de algunas variables o categorías podría llevar a interpretar los resultados de manera errónea o a utilizar métodos de análisis que no sean adecuados. Por esta razón, es importante resolver primero las dudas sobre la estructura, calidad e información del dataset.  

**Punto 10**

|                                                                             Pregunta analitica                                                                            |                        Variables necesarias                        |                                                                                                        ¿Datos suficientes? ¿Por que?                                                                                                       |
|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
| ¿La frecuencia de uso de IA generativa se relaciona  con la confianza para programar y varía  según el programa académico de los estudiantes?                             | uso_ia_generativa,  confianza_programando,  programa               | Sí, con ciertas precauciones. Las variables necesarias  están disponibles. Pero, Antes de comparar programas se  deben estandarizar categorías como “Ciencia Datos” y  “Ciencia de Datos”, y documentar correctamente las escalas.         |
| ¿Los estudiantes que presentan mayores niveles  de estrés tienden a reportar menor  satisfacción académica, y esta relación cambia  según las horas de estudio semanales? | nivel_estres,  satisfaccion_academica,  horas_estudio_semana       | Si, El dataset contiene las tres variables y permite estudiar  la relación entre el estrés, la satisfacción y dedicación  al estudio. Se deben considerar las escalas ordinales y  revisar los valores extremos de horas de estudio antes. |
| ¿En qué medida el tiempo de desplazamiento y  las horas de sueño podrían  estar relacionados con el promedio académico?                                                   | tiempo_desplazamiento_min,  horas_sueno_noche,  promedio_acumulado | Masomenos, ya que aunque las tres variables están disponible,  existen valores faltantes en sueño y desplazamiento, por lo  que primero deberían revisarse y resolver esos espacios.                                                       |

**Punto 11**

La revisión realizada sobre el dataset permitió identificar que contiene 2512 registros y 19 variables relacionadas con características académicas, personales y de hábitos de estudiantes universitarios. La revisión inicial mostró que las variables en términos generales son comprensibles y útiles para realizar análisis posteriores. Sin embargo, se identificaron aspectos que deben corregirse o documentarse antes de continuar, ya que se  encontraron 12 registros duplicados y se detectaron 169 valores faltantes distribuidos entre horas de sueño, tiempo de desplazamiento y área de interés. En las variables categóricas se observaron inconsistencias de escritura, como “Ciencia Datos” frente a “Ciencia de Datos”, “mujer” frente a “Mujer” y “Transmilenio” frente a “TransMilenio”. Además, la fecha de encuesta está almacenada como texto y debería convertirse a formato de fecha. La revisión de rangos identificó valores potencialmente extremos, especialmente 95 horas de estudio semanales y 360 minutos de desplazamiento, que deben ser verificadas antes de decidir si son errores. Finalmente, se recomienda estandarizar categorías, documentar escalas y unidades, revisar duplicados y valores faltantes y validar esos valores extremos que se comentan. Solo después de esto debería realizarse el análisis concreto de los datos.

